test/ex_doc/formatter/html/templates_test.exs

```diff
@@ -109,6 +109,20 @@ defmodule ExDoc.Formatter.HTML.TemplatesTest do
              </h3>
              """
     end
+
+    test "generates headers with unique id's" do
+      assert Templates.link_headings("<h3>Foo</h3>\n<h3>Foo</h3>") == """
+             <h3 id="foo" class="section-heading">
+               <a href="#foo" class="hover-link"><span class="icon-link" aria-hidden="true"></span></a>
+               Foo
+             </h3>
+
+             <h3 id="foo-1" class="section-heading">
+               <a href="#foo-1" class="hover-link"><span class="icon-link" aria-hidden="true"></span></a>
```

Comment:
I would love to see the first one as "#foo" and the second one as "foo-2".

Reply:
What would be the benefit? Personally, I think using the same naming scheme as GitHub/GitLab (see #998) makes sense, but I'm open for suggestions :)

Reply:
The benefit is that we, humans start counting at 1, not 0 as computers do.
I don't know the rationale behind GitHub/GitLab, probably based on the computer language they solved this issue, or because they are just computer geeks.
But if I ever read "#change-2" I don't have to think it will be the third item, and not the second one.
May I ask in reverse, what is the benefit of using a zero-based approach, besides using the same scheme as github/gitlab

Reply:
I agree with the human argument, however, we also like things to be in order, so we'd have to use `#foo-1`, `#foo-2`, etc. which would add a useless suffix to every first (and probably unique) anchor tag.
The way I understood the zero-based approach is that `#foo` is an original heading,  while `#foo-1` is the first reoccurrence of this heading, `#foo-2` is the second [...] and so on.

Reply:
I don't see a problem using "foo" and "foo-2", most of the time it will only be "foo".
"foo" will always come before any "foo-N" if you sort them.

Reply:
I think both formats have their pros and cons. Since the PR is merged, I don't think we will come to a conclusion here. I guess it's best to leave it at that and open a new PR if a change to "foo/foo2" is desired.